doc/application/application.rst

```diff
@@ -675,15 +675,48 @@ following procedure:
 #. Rebuild the application normally following the steps specified
    in `Build an Application`_ above.
 
-
+.. _application_run:
 
 Run an Application
 ******************
 
-An application image can be run on real or emulated hardware. The kernel has
-built-in emulator support for QEMU. It allows you to run and test an application
-virtually, before (or in lieu of) loading and running it on actual target
-hardware.
+An application image can be run on real or emulated hardware.
+
+Most boards supported by Zephyr let you flash a compiled binary using
+the ``flash`` target to copy the binary to the board and run it.
```

Comment:
to clarify, maybe better to say: 

      using the make ``flash`` target

Reply:
Yeah, I was trying to word this in a way that would be compatible with CMake (which supports Ninja in addition to Make), but this section will need to be revisited when that happens anyway.